fedimint-api/src/db/mod.rs

```diff
@@ -169,6 +169,93 @@ impl Drop for CommitTracker {
     }
 }
 
+pub struct ModuleDatabaseTransaction<'a, 'b: 'a> {
+    tx: &'a mut DatabaseTransaction<'b>,
+    module_prefix: Vec<u8>,
+}
+
+impl<'a, 'b> ModuleDatabaseTransaction<'a, 'b> {
+    pub fn new(
+        dbtx: &'a mut DatabaseTransaction<'b>,
+        module_prefix: u16,
+    ) -> ModuleDatabaseTransaction<'a, 'b> {
+        ModuleDatabaseTransaction {
+            tx: dbtx,
+            module_prefix: vec![(module_prefix >> 8) as u8, (module_prefix & 0xff) as u8],
```

Comment:
I thought the prefix is going to be `X <module_id>` where `X` would be some designed prefix for all module data. Otherwise we would have to be careful for the `module_prefix` not to conflict with any other global keys we have. Which is possible maybe if we deem that extra byte not worth it.

Reply:
I added MODULE_GLOBAL_PREFIX (0xff) which is prepended to all module prefixes. As long as the consensus key/values don't use 0xff as a prefix, the data should be separated from the modules.

Reply:
> As long as the consensus key/values don't use 0xff as a prefix

Don't we have like an enum or list of prefixes somewhere? We should reserve it just like any other one we use there.

Reply:
Ah yes, I can add it to the enum of prefixes

Reply:
> Ah yes, I can add it to the enum of prefixes

Actually, it looks like that enum is in fedimint_server, not fedimint_api where the IsolatedDatabaseTransaction impl is.

Maybe we should move the enum from fedimint_server/src/db.rs to fedimint_api?

Reply:
Maybe we could nest them somehow?

Reply:
```

#[repr(u8)]
#[derive(Clone, Debug)]
pub enum DbKeyPrefix {
    ProposedTransaction = 0x01,
    AcceptedTransaction = 0x02,
    DropPeer = 0x03,
    RejectedTransaction = 0x04,
    EpochHistory = 0x05,
    Common(DbKeyPrefixCommon),
}


#[repr(u8)]
#[derive(Clone, Debug)]
pub enum DbKeyPrefixCommon {
    Module = 0xff
}
```

Haven't checked if `EnumIter` is going to complain, but I would throw it away if it does and impl whatever it does manually.

Reply:
THough I haven't checked if this deserializes to u8...

Reply:
Or just move to api, or define some as consts in api, and as enum in server

Reply:
Kept the constant in api but added a new entry in fedimint-server that references that constant.